joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/AstCreatorBase.scala

```diff
@@ -122,6 +122,42 @@ abstract class AstCreatorBase(filename: String) {
     }
   }
 
+  def whileAst(
+    condition: Option[Ast],
+    body: Seq[Ast],
+    code: Option[String] = None,
+    lineNumber: Option[Integer] = None,
+    columnNumber: Option[Integer] = None
+  ): Ast = {
+    var whileNode = NewControlStructure()
+      .controlStructureType(ControlStructureTypes.WHILE)
+      .lineNumber(lineNumber)
+      .columnNumber(columnNumber)
+
+    if (code.isDefined)
```

Comment:
Ideally there would be a `code(Option[String])` method like we have for lines/columns.